src/librustc_trans/trans/_match.rs

```diff
@@ -630,8 +630,7 @@ fn bind_subslice_pat(bcx: Block,
     let vec_datum = match_datum(val, vec_ty);
     let (base, len) = vec_datum.get_vec_base_and_len(bcx);
 
-    let slice_byte_offset = Mul(bcx, vt.llunit_size, C_uint(bcx.ccx(), offset_left));
-    let slice_begin = tvec::pointer_add_byte(bcx, base, slice_byte_offset);
+    let slice_begin = InBoundsGEP(bcx, base, &[C_uint(bcx.ccx(), offset_left)]);
```

Comment:
@huonw Rust code *can* care about addresses of zero-sized-type pointers, however the whole standard library assumes that it doesn't. It's inserting fake values everywhere, it often casts some numerical values to such a pointer. E.g. like that:

    mem::transmute(1us)